### PR TITLE
refactor: Integrate seed recreation flow in btc functions

### PR DESCRIPTION
**Pull request by kartikCypherock**
Implements https://app.clickup.com/t/9002019994/PRF-4827

### CODE REVIEW

File: apps/btc_app/get_public_key.c [hunk 1]
```diff
@@ -248,40 +247,43 @@ static void send_public_key(const uint8_t *public_key) {
  *****************************************************************************/
 
 void btc_get_public_key(btc_query_t *query) {
-  const char *wallet_name = "";
-  char string[100] = "";
-  uint8_t wallet_index = 0;
+  char wallet_name[16] = "";
```
Comment (ujjwal-cyph) [inline apps/btc_app/get_public_key.c:250]: `NAME_SIZE` macro can be used :)

**Review by ujjwal-cyph — APPROVED** [review]
LGTM

File: apps/btc_app/get_public_key.c (hunk 1 above)
Comment (kartikCypherock) [reply apps/btc_app/get_public_key.c]: Thanks for the catch

File: apps/btc_app/get_public_key.c [hunk 2]
```diff
@@ -248,40 +247,43 @@ static void send_public_key(const uint8_t *public_key) {
...
   uint8_t public_key[65] = {0};
   btc_get_public_key_intiate_request_t *init_req =
       &query->get_public_key.initiate;
 
   if (!check_which_request(query, BTC_GET_PUBLIC_KEY_REQUEST_INITIATE_TAG) ||
-      !validate_request_data(&query->get_public_key)) {
+      !validate_request_data(&query->get_public_key) ||
+      !get_wallet_name_by_id(query->get_xpubs.initiate.wallet_id,
+                             (uint8_t *)wallet_name)) {
     return;
```
[the remark this answers is not in the document]
Comment (kartikCypherock) [reply apps/btc_app/get_public_key.c:263]: Added necessary TODOs

**Review by ujjwal-cyph — APPROVED** [review]

**Pull request merged**